crates/hyperdrive-math/src/lp.rs

```diff
@@ -189,15 +334,182 @@ impl State {
+                            let starting_present_value = state.calculate_present_value(current_block_timestamp);
+                            let ending_present_value = new_state.calculate_present_value(current_block_timestamp);
+                            assert!(ending_present_value < starting_present_value);
+                        }
+
+                        else if message == "MinimumTransactionAmount: Not enough lp shares minted." {
+                            let share_contribution =
+                                I256::try_from(contribution / state.vault_share_price()).unwrap();
+                            let new_state = state.get_state_after_liquidity_update(share_contribution);
+                            let starting_present_value = state.calculate_present_value(current_block_timestamp);
+                            let ending_present_value = new_state.calculate_present_value(current_block_timestamp);
+                            let lp_shares = (ending_present_value - starting_present_value)
+                                .mul_div_down(state.lp_total_supply(), starting_present_value);
+                            assert!(lp_shares < state.minimum_transaction_amount());
+                        }
+
+                        else if message == "OutputLimit: Not enough lp shares minted." {
+                            let share_contribution =
+                                I256::try_from(contribution / state.vault_share_price()).unwrap();
+                            let new_state = state.get_state_after_liquidity_update(share_contribution);
+                            let starting_present_value = state.calculate_present_value(current_block_timestamp);
+                            let ending_present_value = new_state.calculate_present_value(current_block_timestamp);
+                            let lp_shares = (ending_present_value - starting_present_value)
+                                .mul_div_down(state.lp_total_supply(), starting_present_value);
+                            assert!(contribution.div_down(lp_shares) < min_lp_share_price);
+                        }
+                    }
+                },
+                // ignore inner panics
+                Err(_) => {}
+            }
+        }
+
+        Ok(())
+    }
+    #[tokio::test]
+    async fn fuzz_test_calculate_add_liquidity() -> Result<()> {
+        // Spawn a test chain and create two agents -- Alice and Bob.
+        let mut rng = thread_rng();
+        let chain = TestChain::new(2).await?;
+        let (alice, bob) = (chain.accounts()[0].clone(), chain.accounts()[1].clone());
+        let mut alice =
+            Agent::new(chain.client(alice).await?, chain.addresses().clone(), None).await?;
+        let mut bob = Agent::new(chain.client(bob).await?, chain.addresses(), None).await?;
+        let config = bob.get_config().clone();
+
+        // Test happy paths.  Limit this to 100 runs as it takes a long time.
+        for _ in 0..100 {
```

Comment:
Looks like we can use `FUZZ_RUNS`, which is set to `100` for longer tests like this (vs `FAST_FUZZ_RUNS` which is `10_000`).